src/metabase/query_processor/middleware/permissions.clj

```diff
@@ -54,7 +54,7 @@
                                     :card-id card-id})))]
       (log/tracef "Required perms to run Card: %s" (pr-str (mi/perms-objects-set card :read)))
       (when-not (mi/can-read? card)
-        (throw (perms-exception (tru "You do not have permissions to view Card {0}." card-id)
+        (throw (perms-exception (tru "You do not have permissions to view Card {0}." (pr-str card-id))
```

Comment:
Unrelated but error messages like `You do not have permissions to view Card 1,234` annoy me so I fixed it so they now read like `You do not have permissions to view Card 1234`